pyseir/load_data.py

```diff
@@ -508,44 +503,185 @@ def load_hospitalization_data_by_state(
-                    * params["hospitalization_length_of_stay_icu"]
-                    + params["hospitalization_rate_icu"]
-                    * params["fraction_icu_requiring_ventilator"]
-                    * params["hospitalization_length_of_stay_icu_and_ventilator"]
-                ) / (params["hospitalization_rate_general"] + params["hospitalization_rate_icu"])
-            else:
-                average_length_of_stay = (
-                    (1 - params["fraction_icu_requiring_ventilator"])
-                    * params["hospitalization_length_of_stay_icu"]
-                    + params["fraction_icu_requiring_ventilator"]
-                    * params["hospitalization_length_of_stay_icu_and_ventilator"]
-                )
-
-            # Now compute a cumulative sum, but at each day, subtract the discharges from the previous count.
-            new_hospitalizations = np.append([0], np.diff(cumulative))
-            current = [0]
-            for i, new_hosps in enumerate(new_hospitalizations[1:]):
-                current.append(current[i] + new_hosps - current[i] / average_length_of_stay)
-            return times_new, current, HospitalizationDataType.CURRENT_HOSPITALIZATIONS
+    if len(df) == 0:
+        return None, None
+
+    # If data available in current_{} column, then return latest not-null value
+    if (df[f"current_{category}"] > 0).any():
+        df = df[df[f"current_{category}"].notnull()]
+        df_latest = df[f"current_{category}"].values.clip(min=0)[-1]
+        times_new = (df["date"].dt.date - t0.date()).dt.days.values
+        times_new_latest = times_new[-1]
+        return times_new_latest, df_latest  # Return current since available
+
+    # If data is available in cumulative, try to convert to current (not just daily)
+    elif (df[f"cumulative_{category}"] > 0).any():
+        # Remove Null & Enforce Monotonically Increasing Cumulatives
+        df = df[df[f"cumulative_{category}"].notnull()]
+        cumulative = df[f"cumulative_{category}"].values.clip(min=0)
+        for i, val in enumerate(cumulative[1:]):
+            if cumulative[i] > cumulative[i + 1]:
+                cumulative[i] = cumulative[i + 1]
+
+        # Estimate Current from Derived Dailies
+        if len(cumulative) >= MIN_DATAPOINTS_TO_CONVERT:
+            current_latest = estimate_current_from_cumulative(cumulative, category)
+            times_new = (df["date"].dt.date - t0.date()).dt.days.values
+            times_new_latest = times_new[-1]
+            return times_new_latest, current_latest  # Return current estimate from cumulative
         else:
-            return times_new, cumulative, HospitalizationDataType.CUMULATIVE_HOSPITALIZATIONS
+            return None, None  # No current, not enough cumulative
     else:
-        return None, None, None
+        return None, None  # No current nor cumulative
+
+
+def estimate_current_from_cumulative(cumulative, category):
```

Comment:
This is excellent. I'm not sure any of this code should live in the `load_data` module though -- maybe we should split out a separate module for ICU estimation, even. 

That can be done in a future PR though.

Reply:
Agreed that it should be pulled out of load_data and that it should be a separate PR.